pennylane/templates/tensornetworks/mps.py

```diff
@@ -17,68 +17,82 @@
         )
 
+    n_step = offset
+    n_layers = len(wires) - int(len(wires) % (n_block_wires // 2)) - n_step
+
     return tuple(
         tuple(wires[idx] for idx in range(j, j + n_block_wires))
         for j in range(
             0,
-            len(wires) - int(len(wires) % (n_block_wires // 2)) - n_block_wires // 2,
-            n_block_wires // 2,
+            n_layers,
+            n_step,
         )
+        if not j + n_block_wires > len(wires)
     )
 
 
 class MPS(Operation):
-    """The MPS template broadcasts an input circuit across many wires following the architecture of a Matrix Product State tensor network.
+    r"""The MPS template broadcasts an input circuit across many wires following the architecture of a Matrix Product State tensor network.
     The result is similar to the architecture in `arXiv:1803.11537 <https://arxiv.org/abs/1803.11537>`_.
 
-    The argument ``block`` is a user-defined quantum circuit.``block`` should have two arguments: ``weights`` and ``wires``.
-    For clarity, it is recommended to use a one-dimensional list or array for the block weights.
+    The keyword argument ``block`` is a user-defined quantum circuit that should accept two arguments: ``wires`` and ``weights``.
+    The latter argument is optional in case the implementation of ``block`` doesn't require any weights. Any additional arguments
+    should be provided using the ``kwargs``.
 
     Args:
         wires (Iterable): wires that the template acts on
         n_block_wires (int): number of wires per block
         block (Callable): quantum circuit that defines a block
         n_params_block (int): the number of parameters in a block; equal to the length of the ``weights`` argument in ``block``
         template_weights (Sequence): list containing the weights for all blocks
+        offset (wires): offset value for positioning the subsequent blocks relative to each other.
+            If ``None``, it defaults to :math:`\text{offset} = \lfloor \text{n_block_wires}/2  \rfloor`,
+            otherwise :math:`\text{offset} \in [1, \text{n_block_wires} - 1]`
+        **kwargs: additional keyword arguments for implementing the ``block``
 
     .. note::
 
-        The expected number of blocks can be obtained from ``qml.MPS.get_n_blocks(wires, n_block_wires)``.
-        The length of ``template_weights`` argument should match the number of blocks.
+        The expected number of blocks can be obtained from ``qml.MPS.get_n_blocks(wires, n_block_wires, offset=0)``, and
+        the length of ``template_weights`` argument should match the number of blocks. Whenever either ``n_block_wires``
+        is odd or ``offset`` is not :math:`\lfloor \text{n_block_wires}/2  \rfloor`, the template deviates from the maximally
+        unbalanced tree architecture described in `arXiv:1803.11537 <https://arxiv.org/abs/1803.11537>`_.
```

Comment:
I think it's can still be represented by a maximally unbalanced tree :thinking: Not a blocker to merging.

Reply:
If you look at the circuit DAG, I don't think so, [unless](https://www.youtube.com/watch?v=A-RfHC91Ewc).